edgelet/build/windows/util.ps1

```diff
@@ -1,57 +1,165 @@
+    Join-Path -Path $ProjectRoot -ChildPath "edgelet/Cargo.toml"
+}
+
+function Get-EdgeletFolder
+{
+    $ProjectRoot = Join-Path -Path $PSScriptRoot -ChildPath "../../.."
+    Join-Path -Path $ProjectRoot -ChildPath "edgelet"
+}
+
+function Get-IotEdgeFolder
+{
+    # iotedge is parent folder of edgelet
+    Join-Path -Path $(Get-EdgeletFolder) -ChildPath ".."
+}
+
+function Assert-Rust
+{
+Param(
+    [Switch] $Arm
+)
+    if($Arm)
+    {
+        if(-NOT (Test-Path "rust-windows-arm"))
+        {
+            # if the folder rust-windows-arm exists, we assume the private rust compiler for arm is installed
+            InstallWinArmPrivateRustCompiler
+        }
+    }
+    elseif (-not (Test-RustUp))
+    {
+        Write-Host "Installing rustup and stable-x86_64-pc-windows-msvc Rust."
+        Invoke-RestMethod -usebasicparsing 'https://static.rust-lang.org/rustup/dist/i686-pc-windows-gnu/rustup-init.exe' -outfile 'rustup-init.exe'
+        if ($LastExitCode)
+        {
+            Throw "Failed to download rustup with exit code $LastExitCode"
+        }
+
+        Write-Host "Running rustup-init.exe"
+        ./rustup-init.exe -y --default-toolchain stable-x86_64-pc-windows-msvc
+        if ($LastExitCode)
+        {
+            Throw "Failed to install rust with exit code $LastExitCode"
+        }
+    }
+    else
+    {
+        Write-Host "Running rustup.exe"
+        rustup install stable-x86_64-pc-windows-msvc
+        if ($LastExitCode)
+        {
+            Throw "Failed to install rust with exit code $LastExitCode"
+        }
+    }
+}
+
+function InstallWinArmPrivateRustCompiler
+{
+    $link = "https://iottools.blob.core.windows.net/iotedge-armtools/rust-windows-arm.zip"
+
+    Write-Host "Downloading $link"
+    $ProgressPreference = 'SilentlyContinue'
+    Invoke-WebRequest $link -out "rust-windows-arm.zip" -UseBasicParsing
+
+    Write-Host "Extracting $link"
+    Expand-Archive -Path "rust-windows-arm.zip" -DestinationPath "rust-windows-arm"
+    $ProgressPreference = 'Stop'
+}
+
+function PatchRustForArm
+{        
+    # arm build requires cl.exe from vc tools to expand a c file for openssl-sys, append x64-x64 cl.exe folder to PATH
+    # $vspath = Join-Path -Path ${env:ProgramFiles(x86)} -ChildPath "Microsoft Visual Studio"
+    # Write-Host $vspath
+    # $cls = (Get-ChildItem -Path $vspath -Filter cl.exe -Recurse -ErrorAction Continue -Force | Sort-Object -Property DirectoryName -Descending)
+    # $clPath = ""
+    # for ($i= 0; $i -lt $cls.length; $i++) {
+
+    #     $cl = $cls[$i]
+    #     # Write-Host $cl.DirectoryName
+    #     if($cl.DirectoryName.ToLower().Contains("hostx64\x64"))
+    #     {
+    #         $clPath = $cl.DirectoryName
+    #         break
+    #     }
+    # }
+
+    # Write-Host $clPath
+
+    # $env:PATH = $clPath + ";" + $env:PATH
+
+    $clpath1 = "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x64;"

```

Comment:
These paths might break if the VSTS VMs / build VMs update to a newer VS, no? Can you discover the path using the loop you commented out above?

Also, can you add them only if `cl.exe` doesn't already exist in `PATH` ? ie do the `Get-Command` test first.